microsoft/PowerToys
# tweak plugin settings

## Summary of the Pull Request

**What is this about:**
It is preparation for #5273. At first, we need to take plugins settings from `PowerToys Run\settings.json` so they can be edited from the settings app. The settings watcher needs to propagate changes correctly so they are used by PT Run right away. Also, some refactoring was done.

**What is include in the PR:** 
- Add Plugins array property to `PowerToys Run\settings.json`. If the property does not exist it is created and populated with default plugins settings
- Get rid of `PluginsSettings.cs` and update `PluginManager.AllPlugins` from `SettingsWatcher`
- Get rid of action keywords in plugin.json files in favour of `ActionKeyword` and `IsGlobal`

**How does someone test / validate:** 
- Verify if it is consistent with old settings. 
- Delete AppData settings, start PT and check that PT Run works
- Trigger every plugin through global query as well as using action keyword
- Try to update `PowerToys Run\settings.json` and see if settings are populated to PT Run

We should test it well for regression!

## Quality Checklist

- [X] **Linked issue:** #9488

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
